src/ui/widgets/tui_topbar.rs

```diff
@@ -31,8 +31,21 @@ impl<'a> Widget for TuiTopBar<'a> {
 
         if curr_path_str.len() > area.width as usize {
             if let Some(s) = self.path.file_name() {
+                /* path shortener */
+                let mut temp: Vec<&str> = curr_path_str.split("/").collect();
```

Comment:
Instead of using split here, I think we can use [Path.components](https://doc.rust-lang.org/std/path/struct.Path.html#method.components) method on `self.path` instead.
This will work on windows as well (not that joshuto supports windows currently), but it will also eliminate the problem of needing to check if the folder name is not empty string.

Something like:
```rust
let mut short_path = String::new();
for component in self.path.components() {
    match component {
        Component::RootDir => short_path.push('/')
        Component::Normal(s) => {
            // get first char and push it
            // push "/"
        }
    }
}
```

Reply:
neat thinking!